### DISCUSSION TITLE
Multiple template content areas / editable and locked content

### DISCUSSION THREAD

**Discussion by sf-steve** [elementor/elementor#21309] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Since the advent of WYSIWYG wordpress editors, we (as a dev shop) have always found a conflict between design flexibility, and control.
For the longest time, control took precedent, with themes and sites developed with hard coded templates, and user editing handled with meta boxes, then later ACF fields. As well as design control, this approach ensured 100% control for extendibility and feature development (for us, the agency). Most early WYSIWYG plugins frankly sucked, with awful end user UI, poor or non existent developer docs, and limited extension points.

Elementor changed that. Excellent developer docs gave us the confidence we would never be unable to develop a feature due to the WYSIWYG tool. and the end user UI is very intuitive.

Once familiar with the tool, our design team can directly handle the majority of front end development, and can iterate instantly. Our marketing team can spin up landing pages and sales funnels from existing saved templates and sections, again with limited or zero interaction with design and dev teams.

It has fundamentally changed the way we work on wordpress CMS projects, making us far more efficient, more profitable, and able to provide better value for our clients.

However, the conflict between control and flexibility still remains. Currently, we handle this differently depending on the size of the project and clients tech team.
For small projects, those where we will handle the majority of the site editing, or the client has a small editing team we can train, we go with Elementor pretty much everywhere.

For larger projects, with larger client editing teams, we follow a more locked down approach, with ACF and templates.

However, we have always wished we could have the best of both worlds - a locked down editing experience, but allowing the end user the power of Elementor for those areas which would benefit from it.

The current template system allows for us to design a template for a certain post type, and have a single editable area (via a post-content widget), plus non WYSIWYG content via dynamic fields.

What it does not allow us to do, it have multiple WYSIWYG editable areas, or have the end user edit the page in a single screen (any dynamic content must be edited on the regular edit screen)

The advent of the new Tabs widget and continued work to add nested widgets has made us evaluate this process again, hence this discussion post.

### Describe the solution you'd like

There are a couple of approaches we can envision for handling this.

The cleanest approach, IMO, would be some form of `content_area` widget, that can be added to Elementor post type templates (assigned via theme builder).

On creating / editing a post of said type with elementor, the end user would be able to add widgets into those areas. A very similar UI to the current experience, but with multiple content areas, defined by the template.

This seems like the simplest to implement technically, as the conceptual model would remain similar. The meta data for the template would remain attached to the post_id of the template, and the meta_data for the post would remain attached to the post.

Another option would be a lockable/editable regions feature. To explain, i need to list my understanding of templates within the current system.

Currently, templates can be used in 3 different ways (_side note, some work on verbiage is probably warranted, as having very different things with the same name can be confusing_).

1. in a similar way to coded theme templates - assigned to a certain location (header, footer, single post etc) via theme builder, and pulling their content from the individual post via dynamic fields and post-content, on the front end. 
2. As a saved layout that can be copied into a page, with any edits made on the copy, saved to the new pages meta data, and completely unlinked from the original template. For clarity, i will refer to this type of template as a saved layout.
3. As a saved layout, that can be directly embeded into another page with the template widget, with any changes made on the original reflected wherever it is embeded.

In addition, there is also the concept of a global widget. This is something we have never really found a use for in its current form, seemingly offering a limited subset of the features available via the template widget outlined in #3 above.

The idea here is that saved-layouts can be assigned editable and or locked regions. On loading into a page, the user would only be able to edit the editable parts. To be really useful however, it needs to maintain a link to the original saved-layout, so that future changes to said layout can be reflected throughout the site, therefor acting as something of a hybrid between #2 & #3 above.

This approach is likely far more challenging technically, and i'm not sure if offers anything that the first approach (multiple `content_areas` within a template) doesn't.

### Describe alternatives you've considered

It is possible to build similar functionality with custom post types - essentially having every editable area be a separate post of some custom type, and pulling it all together on the front end.
However, this provides a terrible editing UX.

This kind of approach has been taken by a number of 3rd party plugins to overcome the nested widget limitation. These will soon become obsolete with the advent of nested widgets in core. It is my hope that we can achieve the same thing here, removing the trade-off between the flexibility and UX of Elementor, and the control & long term maintainability of a more locked down coded solution.

**Comment by baf139** [comment]
I would love this. 
to simplify. We want to have certain parts of the single post template locked down to fields from ACF but the ability for users to add containers/widgets to the particular post they are editing. 

The best example would be that we have a listing site and we want to be able to add a gallery using frontend elementor widgets vs. having to use ACF pro or Dynamic content plugins. The gallery lives in the sidebar but the provided "Post Content" lives in the body container. 

workflow would be:
user creates new post
user fills out all fields in ACF
user clicks edit with elementor
user uses widgets to polish off the post.

The problem is that you can only drag new widgets to 1 designated "Post Content" section. We would like to have a predefined set of "Content Placeholders" that users could add per post content to while in the elementor editing experience.
